featurebyte/models/feature.py

```diff
@@ -217,20 +217,28 @@ class FeatureModel(FeatureByteCatalogBaseDocumentModel):
     @root_validator
     @classmethod
     def _add_derived_attributes(cls, values: dict[str, Any]) -> dict[str, Any]:
-        # extract table ids & entity ids from the graph
-        graph = values["graph"]
-        node_name = values["node_name"]
-        values["primary_table_ids"] = graph.get_primary_table_ids(node_name=node_name)
-        values["table_ids"] = graph.get_table_ids(node_name=node_name)
-        values["entity_ids"] = graph.get_entity_ids(node_name=node_name)
-
-        # extract dtype from the graph
-        node = graph.get_node_by_name(node_name)
-        op_struct = graph.extract_operation_structure(node=node)
-        if len(op_struct.aggregations) != 1:
-            raise ValueError("Feature graph must have exactly one aggregation output")
-
-        values["dtype"] = op_struct.aggregations[0].dtype
+        derived_attributes = [
+            values.get("primary_table_ids"),
+            values.get("table_ids"),
+            values.get("entity_ids"),
+            values.get("dtype"),
+        ]
+        if any(not x for x in derived_attributes):
```

Comment:
Would it be more correct to check for `x is None` since some attributes like `entity_ids` can be an empty list in some cases?

Reply:
`x is None` is incorrect as some attributes could be initialized with `[]` and we want to initialize it with root validator logic.

Reply:
One of the options is to skip checking `entity_ids`, wdyt?

Reply:
Then it seems fine to me to keep this as is, the cost is that loading features without entities will be slightly more expensive because of the extra derivation here, but that seems acceptable.